scripts/tests/test_zkai_native_attention_mlp_single_proof_route_gate.py

```
@@ -0,0 +1,122 @@
+            self.assertEqual(loaded_raw, raw)
+        finally:
+            path.unlink(missing_ok=True)
+
+    @unittest.skipUnless(hasattr(os, "symlink"), "symlink support required")
+    def test_source_artifact_read_rejects_symlink_source(self) -> None:
+        target = gate.EVIDENCE_DIR / ".tmp-native-attention-mlp-route-target.json"
+        link = gate.EVIDENCE_DIR / ".tmp-native-attention-mlp-route-link.json"
+        try:
+            target.write_text('{"ok": true}\n', encoding="utf-8")
+            link.symlink_to(target)
+            with self.assertRaisesRegex(gate.NativeAttentionMlpSingleProofRouteError, "symlink"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)